File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        for (int i = 0; i < nums.size(); i++) {
            //  cout<<"this is nums"<<nums[i]<< "and target"<<" "<< target;
            if (nums[i] >= target) {
               
                return i;
            }
        }

        return nums.size();
    }
};

File: 0035-search-insert-position/0035-search-insert-position.ts
function searchInsert(nums: number[], target: number): number {
    let low=0 ;
    let high=nums.length;
    
    while(low<high){
        let mid = low + Math.floor((high - low) / 2);

        if(nums[mid]>=target){
            high= mid;
        } 
        else {
            low= mid+1;
        }
    }

    return low;
};